libbeat/logp/file_rotator.go

```diff
@@ -73,16 +75,26 @@ func (rotator *FileRotator) WriteLine(line []byte) error {
 	}
 
 	line = append(line, '\n')
+
+	rotator.currentLock.RLock()
 	_, err := rotator.current.Write(line)
```

Comment:
Shouldn't the `Write` and the increment of the `currentSize` occur within the same lock block? Without that it seems that the `currentSize` may not be accurate when it is checked by other goroutines.